### PR TITLE
Feature/profiles

### PR DESCRIPTION
**Pull request by cmanou**

### CODE REVIEW

File: src/main/scala/com/ureferjobs/oden/models/server/Link.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.ureferjobs.gaia.mongo.record._
+import net.liftweb.common.Box
+import net.liftweb.mongodb.record._
+import net.liftweb.record.field.StringField
+
+/**
+ * Universal link type
+ */
+class Link extends BsonRecord[Link] {
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/models/server/Link.scala]: @cmanou this class requires some specs

File: src/main/scala/com/ureferjobs/oden/services/Responder.scala [hunk 2]
```diff
@@ -31,6 +31,7 @@ class Responder extends Service[Request, Response] with Logger {
         // User RESTful actions
         case GET  -> Root / "users" / userId => users.ShowResponder(request, Map("id" -> userId))
         case POST -> Root / "users" => users.CreateResponder(request, Map())
+        case PUT  -> Root / "users" / userId => users.UpdateResponder(request, Map("id" -> userId))
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/services/Responder.scala]: @cmanou this was added as a result of profiles, so we should spec this please

File: src/main/scala/com/ureferjobs/oden/services/Responder.scala [hunk 3]
```diff
@@ -38,12 +39,12 @@ class Responder extends Service[Request, Response] with Logger {
         case PUT  -> Root / "users" / userId / "resetpassword" => users.ResetPasswordResponder(request, Map("id" -> userId))
 
         // Company RESTful actions
+        case GET  -> Root / "companies" / companyId => companies.ShowResponder(request, Map("id" -> companyId))
         case GET  -> Root / "companies" => companies.IndexResponder(request, Map())
         case POST -> Root / "companies" => companies.CreateResponder(request, Map())
         case PUT  -> Root / "companies" / companyId => companies.UpdateResponder(request, Map("id" -> companyId))
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/services/Responder.scala]: @cmanou same here, as "profiles" for companies are allowed we should spec it too.

File: src/main/scala/com/ureferjobs/oden/services/responders/companies/ShowResponder.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import net.liftweb.json._
+
+/**
+ * The service returns the list of all active roles in the system
+ */
+object ShowResponder extends GeneralResponder {
+  def apply(request: Request, params: Map[String, String]): Future[Response] = Future.value {
+    findCompany(params.get("id")) match {
+      case Full(company) =>
+        val data = pretty(render(company.asJValue))
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/services/responders/companies/ShowResponder.scala]: @cmanou we can use compactRender here

File: src/main/scala/com/ureferjobs/oden/services/responders/users/CreateResponder.scala [hunk 5]
```diff
@@ -20,10 +20,12 @@ object CreateResponder extends Logger {
       // No users match the email and the de-serialised user object is valid
       case 0 if user.validate.isEmpty =>
         user.save()
-        Response(Http11, Created)
+        val response = Response(Http11, Created)
+        response.location_=(user.identifier)
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/services/responders/users/CreateResponder.scala]: @cmanou perhaps the spec should be extended to check to make sure the location is coming back

File: src/main/scala/com/ureferjobs/oden/services/responders/users/UpdateResponder.scala [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import com.ureferjobs.gaia.services.Responses._
+import com.ureferjobs.oden.models.server.Users
+import net.liftweb.common.Full
+import net.liftweb.common.Logger
+import net.liftweb.json._
+
+/**
+ * The service returns the list of all active roles in the system
+ */
+object UpdateResponder extends GeneralResponder with Logger {
```
Comment (shirren) [inline src/main/scala/com/ureferjobs/oden/services/responders/users/UpdateResponder.scala]: @cmanou missing specs

File: src/main/scala/com/ureferjobs/oden/services/responders/users/CreateResponder.scala (hunk 5 above)
Comment (cmanou) [inline src/main/scala/com/ureferjobs/oden/services/responders/users/CreateResponder.scala]: done

File: src/main/scala/com/ureferjobs/oden/services/Responder.scala (hunk 3 above)
Comment (cmanou) [inline src/main/scala/com/ureferjobs/oden/services/Responder.scala]: done

File: src/main/scala/com/ureferjobs/oden/services/Responder.scala (hunk 2 above)
Comment (cmanou) [inline src/main/scala/com/ureferjobs/oden/services/Responder.scala]: done

**Comment by shirren** [comment]
@cmanou I changed the model slightly to include a CompanyProfile. I think it makes sense to make the User and Company models somewhat similar for consistency.

**Pull request merged**